lib/evaluation/sameSubjectEvaluationHandler.ts

```diff
@@ -39,9 +39,9 @@ export class SameSubjectEvaluationHandler extends AbstractEvaluationHandler {
     return 'SameSubjectEvaluationHandler';
   }
 
-  public handle(pd: PresentationDefinition, vp: VerifiablePresentation): void {
+  public handle(pd: PresentationDefinition): void {
     this.pDefinition = pd;
-    this.vPresentation = vp;
+    this.vPresentation = this.client.verifiablePresentation;
```

Comment:
The verifiable presentation that gets handled here is the one created in step 3. I'm just wondering if that is correct, because it violates the contract that the same verifiable presentation should be evaluated in each step. Maybe this step does not fit the chain?

Reply:
because of different entry points of our steps (`selectFrom` & `evaluate`) we need to make this change, because we might have only `verifiableCredential`s in the first step. and needless to say the `presentation_submission` object is created based on the `presentationDefinition` object that we are passing to our chain.

Reply:
This is okay: As we have PresentationDefinition, VerifiablePresentation, AlteredVerifiablePresentaiton, Results. And based on these 4 things we proceed.

So since even conseptually the VP and AlteredVP are differrent. This is ok.
Missing param does look like it is violating the DesignPattern. But this param is absent just because we are in JS and we can remove it.